### PR TITLE
test: Created tests for RegistrationVerificationPresenter

### LINKED ISSUE #495: Unit tests for RegistrationVerificationPresenter
**Summary:**
Write Unit tests for RegistrationVerificationPresenter.

Example for unit testing of <a href="https://github.com/openMF/mifos-mobile/blob/development/app/src/main/java/org/mifos/mobilebanking/presenters/AccountsPresenter.java"> AccountPresenter</a> can be seen in <a href="https://github.com/openMF/mifos-mobile/blob/development/app/src/test/java/org/mifos/mobilebanking/AccountsPresenterTest.java"> AccountsPresenterTest</a>

### PR DESCRIPTION
**Pull request by se-ke**
fixes #495

### CODE REVIEW

File: app/src/main/java/org/mifos/mobilebanking/presenters/RegistrationVerificationPresenter.java [hunk 1]
```diff
@@ -69,7 +70,7 @@ public void onCompleted() {
                     @Override
                     public void onError(Throwable e) {
                         getMvpView().hideProgress();
-                        getMvpView().showError(MFErrorParser.errorMessage(e));
+                        getMvpView().showError(context.getString(R.string.fail_verify));
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/presenters/RegistrationVerificationPresenter.java]: Do not show a custom error msg by using context.getString(R.string.fail_register), as MFErrorParser.errorMessage(e) contains important error msg which should be displayed to the user so revert back to it.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -154,6 +154,7 @@
     <string name="error_creating_beneficiary">Failed to create Beneficiary</string>
     <string name="error_updating_beneficiary">Failed to update Beneficiary</string>
     <string name="error_deleting_beneficiary">Failed to delete Beneficiary</string>
+    <string name="fail_verify">Failed to Verify</string>
```
Comment (dilpreet2028) [inline app/src/main/res/values/strings.xml]: Remove this string as we don't need custom error msg.

File: app/src/test/java/org/mifos/mobilebanking/RegistrationVerificationPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    @Test
+    public void testVerifyUser() throws Exception {
+        when(dataManager.verifyUser(userVerify)).thenReturn(Observable.just(responseBody));
+
+        presenter.verifyUser(userVerify);
+
+        verify(view).showProgress();
+        verify(view).hideProgress();
+        verify(view).showVerifiedSuccessfully();
+        verify(view, never()).showError(context.getString(R.string.fail_verify));
```
Comment (dilpreet2028) [inline app/src/test/java/org/mifos/mobilebanking/RegistrationVerificationPresenterTest.java]: Have a look at RegistrationPresenterTest is doing it.

File: app/src/test/java/org/mifos/mobilebanking/RegistrationVerificationPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public void testVerifyUserFails() throws Exception{
+        when(dataManager.verifyUser(userVerify)).thenReturn(Observable.<ResponseBody>error(new
+                RuntimeException()));
+
+        presenter.verifyUser(userVerify);
+
+        verify(view).showProgress();
+        verify(view).hideProgress();
+        verify(view, never()).showVerifiedSuccessfully();
+        verify(view).showError(context.getString(R.string.fail_verify));
```
Comment (dilpreet2028) [inline app/src/test/java/org/mifos/mobilebanking/RegistrationVerificationPresenterTest.java]: Have a look at RegistrationPresenterTest is doing it.

**Review by dilpreet2028 — COMMENTED** [review]
Your travis build are also failing so before submitting a PR make sure to perform `./gradlew check` to ensure that build is passing if not then fix the errors. 
And fix the conflicts.

**Comment by se-ke** [comment]
Ok, I addressed your concerns and resolved the conflicts, but I believe I don't have the ability to merge the different pull requests (to fix the conflicts). Also, the third_party_license_metadata and third_party_licenses were automatically updated by Android Studio when I ran checks and such, so they are included in the commits.

**Comment by dilpreet2028** [comment]
It seems like your are working in the development branch . Let us schedule a hangouts call I will explain you how to fix this, ping me at dilpreet2028@gmail.com

**Pull request closed without merging**